### PR TITLE
Farhan Ahmad Nurzi / Task - Refactor containers components [Translation Key Extraction]

### PR DESCRIPTION
**Pull request by farhan-nurzi-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Comment by niloofar-deriv** [comment]
Please resolve the conflicts :pray:

File: src/pages/help-centre/components/_questions-template.tsx [hunk 1]
```diff
@@ -32,10 +32,8 @@ const QuestionsTemplate = ({ data }: TQuestionsTemplate) => {
     return (
         <Layout>
             <SEO
-                title={localize(
-                    `Help centre | Frequently asked questions | ${untranslate_category} | Deriv`,
-                )}
-                description={localize(`Frequently asked questions - ${untranslate_category}`)}
+                title={`_t_Help centre | Frequently asked questions | ${untranslate_category} | Deriv_t_`}
+                description={`_t_Frequently asked questions - ${untranslate_category}_t_`}
```
[the remark this answers is not in the document]
Comment (farhan-nurzi-deriv) [reply src/pages/help-centre/components/_questions-template.tsx:36]: Screenshot 2023-03-13 at 10 36 17 AM
its not translated, is this okay?

File: src/components/containers/seo.tsx [hunk 2]
```diff
@@ -18,22 +19,28 @@ type SiteMetadataType = {
...
+
+type TLocalizeParams = {
+    key: TString
+    values?: { search: string }
+}
+
 type MetaAttributesType = {
-    og_title?: string
-    og_description?: string
+    og_title?: TString | TLocalizeParams
```
Comment (niloofar-deriv) [inline src/components/containers/seo.tsx:29]:
```suggestion
    og_title?: TString
```

File: src/components/containers/seo.tsx [hunk 3]
```diff
@@ -18,22 +19,28 @@ type SiteMetadataType = {
...
+    og_description?: TString | TLocalizeParams
```
Comment (niloofar-deriv) [inline src/components/containers/seo.tsx:30]:
```suggestion
    og_description?: TString
```

File: src/components/containers/seo.tsx [hunk 4]
```diff
@@ -18,22 +19,28 @@ type SiteMetadataType = {
...
     og_type?: string
     og_img?: string
     og_img_width?: string
     og_img_height?: string
 }
 
 type SeoProps = {
-    description?: string
+    description?: TString | TLocalizeParams
```
Comment (niloofar-deriv) [inline src/components/containers/seo.tsx:38]:
```suggestion
    description?: TString
```

File: src/components/containers/seo.tsx [hunk 5]
```diff
@@ -18,22 +19,28 @@ type SiteMetadataType = {
...
     has_organization_schema?: boolean
     meta?: { name: string; content: string | keyof MetaAttributesType }
     meta_attributes?: MetaAttributesType
     no_index?: boolean
-    title?: string
+    title?: TString | TLocalizeParams
```
Comment (niloofar-deriv) [inline src/components/containers/seo.tsx:43]:
```suggestion
    title?: TString
    title_values?: { search: string }
```

File: src/components/containers/seo.tsx [hunk 6]
```diff
@@ -18,22 +19,28 @@ type SiteMetadataType = {
         title?: string
     }
 }
+
+type TLocalizeParams = {
```
Comment (niloofar-deriv) [inline src/components/containers/seo.tsx:23]:
```suggestion
```

File: src/pages/help-centre/components/_questions-template.tsx [hunk 7]
```diff
@@ -32,10 +32,14 @@ const QuestionsTemplate = ({ data }: TQuestionsTemplate) => {
     return (
         <Layout>
             <SEO
-                title={localize(
-                    `Help centre | Frequently asked questions | ${untranslate_category} | Deriv`,
-                )}
-                description={localize(`Frequently asked questions - ${untranslate_category}`)}
+                title={{
```
Comment (niloofar-deriv) [inline src/pages/help-centre/components/_questions-template.tsx:35]:
```suggestion
                title="_t_Help centre | Frequently asked questions | {{untranslate_category}} | Deriv_t_"
                title_values={ search: untranslate_category }
```

File: documents/implementation-guide.md [hunk 8]
```diff
@@ -93,9 +93,9 @@ Usage example:
 ```js
 import { localize, Localize } from 'components/localization'
 
-<p>{localize('text need to be translated')}</p>
+<p>{localize('_t_text need to be translated_t_')}</p>
```
Comment (NikitK-deriv) [inline documents/implementation-guide.md:96]: We have to provide current approach in guide as well

Screenshot 2023-03-13 at 10 30 02

File: src/components/containers/seo.tsx [hunk 9]
```diff
@@ -124,7 +137,13 @@ const SEO = ({
...
                 dir: lang_direction,
             }}
-            title={title}
+            title={
+                title
+                    ? typeof title === 'string'
+                        ? localize(title)
+                        : localize(title.key, title.values)
+                    : null
+            }
```
Comment (NikitK-deriv) [inline src/components/containers/seo.tsx]: title as undefined will no broke application
```suggestion
            title={
                 typeof title === 'string'
                        ? localize(title)
                        : localize(title.key, title.values)
            }
```

File: src/components/containers/seo.tsx [hunk 10]
```diff
@@ -137,15 +156,29 @@ const SEO = ({
...
                     property: 'og:title',
-                    content: meta_attributes?.og_title || default_og_title,
+                    content: meta_attributes?.og_title
+                        ? typeof meta_attributes.og_title === 'string'
+                            ? localize(meta_attributes.og_title)
+                            : localize(
+                                  meta_attributes.og_title.key,
+                                  meta_attributes.og_title.values,
+                              )
+                        : default_og_title,
```
Comment (NikitK-deriv) [inline src/components/containers/seo.tsx]: Please refactor this and the similar things
```suggestion
                    content: (typeof meta_attributes.og_title === 'string'
                            ? localize(meta_attributes.og_title)
                            : localize(
                                  meta_attributes.og_title.key,
                                  meta_attributes.og_title.values,
                              ))
                        || default_og_title,
```

**Review by niloofar-deriv — APPROVED** [review]
LGTM :+1:

**Review by NikitK-deriv — APPROVED** [review]

**Pull request merged**